Selector/Input/styles.scss

```diff
@@ -0,0 +1,470 @@
+  }
+
+  &.three-quarters {
+    @include partial-width-input(75%);
+  }
+
+  &.two-thirds {
+    @include partial-width-input(66.66666%);
+  }
+
+  &.sixty {
+    @include partial-width-input(60%);
+  }
+
+  &.half {
+    @include partial-width-input(50%);
+  }
+
+  &.forty {
+    @include partial-width-input(40%);
+  }
+
+  &.third {
+    @include partial-width-input(33.33333%);
+  }
+
+  &.quarter {
+    @include partial-width-input(25%);
+  }
+
+  &.twenty {
+    @include partial-width-input(20%);
+  }
+
+  &.left {
+    @include respond-to-wide {
+      margin-right: 12.5px;
+    }
+  }
+
+  &.center {
+    @include respond-to-wide {
+      margin: 0 12.5px;
+    }
+
+    &::after {
+      clear: both;
+      content: '';
+      display: block;
+    }
+  }
+
+  &.right {
+    @include respond-to-wide {
+      margin-left: 12.5px;
+    }
+
+    &::after {
+      clear: both;
+      content: '';
+      display: block;
+    }
+  }
+}
+
+%input {
+  @include stackable-input();
+
+  border-bottom: ($grid * .2) solid $border-color;
+  box-sizing: border-box;
+  clear: both;
+  cursor: pointer;
+  display: block;
+  height: ($grid * 10);
+  line-height: ($grid * 10);
+  position: relative;
+  text-align: left;
+  transition: border-color .2s ease;
+
+  &.big {
+    height: ($grid * 12);
+    line-height: ($grid * 12);
+  }
+
+  &.giant {
+    height: ($grid * 16);
+    line-height: ($grid * 16);
+  }
+
+  &.is-hidden {
+    display: none;
+  }
+
+  &:focus {
+    border-bottom-width: ($grid * .4);
+    outline: none;
+  }
+
+  &:hover {
+    border-color: map-get($colors, blue-hover);
+    z-index: 10;
+  }
+
+  &.is-focused {
+    border-bottom-width: ($grid * .4);
+    border-color: map-get($colors, blue-focus);
+    z-index: 20;
+  }
+
+  &.is-error {
+    border-color: map-get($colors, error-border);
+    box-shadow: none;
+    z-index: 5;
+
+    &:hover {
+      border-color: map-get($colors, error-hover);
+      z-index: 10;
+    }
+
+    &.is-focused {
+      border-color: map-get($colors, error);
+      z-index: 20;
+    }
+  }
+
+  &.is-warning {
+    border-color: map-get($colors, warning-border);
+    box-shadow: none;
+    z-index: 5;
+
+    &:hover {
+      border-color: map-get($colors, warning-hover);
+      z-index: 10;
+    }
+
+    &.is-focused {
+      border-color: map-get($colors, warning-hover);
+      z-index: 20;
+    }
+  }
+
+  &.is-disabled {
+    border-color: map-get($colors, grey-lines);
+
+    &:hover {
+      border-color: map-get($colors, grey-lines);
+    }
+  }
+
+  &.is-centered {
+    .input__label {
+      padding: 0;
+      width: 100%;
+    }
+
+    .input--icon__input {
+      padding: 0 40px;
+    }
+
+    .input__input,
+    .input__label {
+      text-align: center;
+    }
+  }
+
+  &.is-loading {
+    &:after {
+      @include button__loader;
+      content: '';
+      margin-top: -10px;
+      position: absolute;
+      right: 15px;
+      top: 50%;
+    }
+  }
+}
+
+%input__label {
```

Comment:
Let's also rename this templates to `%select-input<whatever>`. They are repeated from `input`, but extracting them might be problematic, usually duplicating these works better.